### PR TITLE
Slightly more generic country profiles (+ energy)

### PR DESCRIPTION
**Pull request by mlbrgl**
I cleaned up some of the remaining COVID specific code for country profiles, and made adding more slightly more straightforward (less places to update). 

Most of the heavy lifting previously done by @shaahmad when allowing more than 1 country profile.

Reviews welcome but please do not merge as this requires syncing with WP updates. Thanks!

### CODE REVIEW

File: site/server/SiteBaker.tsx [hunk 1]
```diff
@@ -231,22 +227,11 @@ export class SiteBaker {
...
-
-        // Maxes out resources (TODO: RCA)
-        // await Promise.all(bakingPosts.map(post => this.bakePost(post)))
-
-        // Delete any previously rendered posts that aren't in the database
+    // Returns the slugs of posts which exist on the filesystem but are not in the DB anymore.
+    // This happens when posts have been saved in previous bakes but have been since then deleted, unpublished or renamed.
+    // Among all existing slugs on the filesystem, some are not coming from WP. They are baked independently and should not
+    // be deleted if WP does not list them (e.g. grapher/*).
+    private getPostSlugsToRemove(postSlugsFromDb: string[]) {
```
Comment (breck7) [inline site/server/SiteBaker.tsx:234]: Nice! Much cleaner.

File: site/server/SiteBaker.tsx [hunk 2]
```diff
@@ -274,8 +260,27 @@ export class SiteBaker {
...
+            const post = await wpdb.getFullPost(postApi)
+
+            postSlugs.push(post.slug)
+            await this.bakePost(post)
+        }
+
+        // Maxes out resources (TODO: RCA)
+        // await Promise.all(bakingPosts.map(post => this.bakePost(post)))
+
+        // Delete any previously rendered posts that aren't in the database
```
Comment (breck7) [inline site/server/SiteBaker.tsx:282]: Nit: in addition to your cleanup above, what if we also put this in a separate method? Maybe `deleteStalePosts`? Then have the `bakePosts` caller call this one after bakePosts.

File: adminSite/server/mockSiteRouter.tsx [hunk 3]
```diff
@@ -37,10 +37,9 @@ import {
     covidDashboardSlug,
     covidChartAndVariableMetaPath
 } from "explorer/covidExplorer/CovidConstants"
-import { covidCountryProfileRootPath } from "site/server/covid/CovidConstants"
```
Comment (breck7) [inline adminSite/server/mockSiteRouter.tsx:40]: Nice! Get to delete a whole folder with that one.

File: adminSite/server/mockSiteRouter.tsx [hunk 4]
```diff
@@ -126,24 +125,10 @@ mockSiteRouter.get(`/${covidDashboardSlug}`, async (req, res) => {
...
-const co2Profile = countryProfileSpecs.get("co2")!
-
-mockSiteRouter.get(`/${co2Profile.rootPath}/:countrySlug`, async (req, res) => {
-    res.send(
-        await countryProfileCountryPage(co2Profile, req.params.countrySlug)
-    )
+countryProfileSpecs.forEach(spec => {
+    mockSiteRouter.get(`/${spec.rootPath}/:countrySlug`, async (req, res) => {
+        res.send(await countryProfileCountryPage(spec, req.params.countrySlug))
+    })
```
Comment (shaahmad) [inline adminSite/server/mockSiteRouter.tsx:131]: nice

File: site/server/SiteBaker.tsx (hunk 2 above)
Comment (mlbrgl) [reply site/server/SiteBaker.tsx:282]: right, it is true that the baking of new posts and deletion of stale ones are conceptually two different things. I don't know if I can resolve to breaking the beautiful symmetry of `bakeAll()` though :D

Comment (mlbrgl) [reply site/server/SiteBaker.tsx:282]: for background, I extracted that behaviour into `getPostSlugsToRemove` to make it testable, as this is quite a touchy piece of code, which directly impacts what is made visible from one bake to the next.

Comment (mlbrgl) [reply site/server/SiteBaker.tsx:289]: @breck7 merging this as-is for now, but feel free to resume that conversation.

**Comment by mlbrgl** [comment]
@breck7 @shaahmad thanks for the review!

**Pull request merged**